### PR TITLE
Define `TEST_CATEGORY_DEATH` macro missing for SYCL and HIP

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: core/unit_test/category_files/TestHIP_Category.hpp [hunk 1]
```diff
@@ -49,6 +49,7 @@
 
 #define TEST_CATEGORY hip
 #define TEST_CATEGORY_NUMBER 6
+#define TEST_CATEGORY_DEATH hip_DeathTest
```
Comment (Rombur) [inline core/unit_test/category_files/TestHIP_Category.hpp:52]: It's also missing in HIPHostPinned

**Review by masterleinad — CHANGES_REQUESTED** [review]
Also missing for the other `SYCL` memory spaces.

File: core/unit_test/category_files/TestHIP_Category.hpp (hunk 1 above)
Comment (dalg24) [reply core/unit_test/category_files/TestHIP_Category.hpp:52]: I don;'t think the macro gets used for auxiliary spaces but sure I will add

**Review by masterleinad — APPROVED** [review]
Thanks!

**Review by crtrott — APPROVED** [review]

**Pull request merged**